.github/workflows/build.yml

```diff
@@ -40,8 +40,8 @@ jobs:
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
       - uses: actions/setup-go@f111f3307d8850f501ac008e886eec1fd1932a34 # v5.3.0
         with:
-          go-version-file: 'go.mod'
-          cache: true
+          go-version: '1.23'
```

Comment:
why are we pinning the go version
would it be easier to just update the go.mod to 1.23.5?

Reply:
we can that as well i am not a big fan but this is not being used as a lib, so does not matter, will update the go.mod

Reply:
PTAL